src/ui/layout/mct-tree.vue

```diff
@@ -125,28 +136,71 @@
 
 <script>
     import treeItem from './tree-item.vue'
+    import search from '../components/search.vue';
 
     export default {
-        data() {
-            return {
-                children: []
-            };
-        },
         inject: ['openmct'],
-        mounted: function () {
-            this.openmct.objects.get('ROOT')
-                .then(root => this.openmct.composition.get(root).load())
-                .then(children => this.children = children.map((c) => {
-                    return {
-                        id: this.openmct.objects.makeKeyString(c.identifier),
-                        object: c,
-                        objectPath: [c]
-                    };
-                }))
-        },
         name: 'mct-tree',
         components: {
+            search,
             treeItem
+        },
+        data() {
+            return {
+                searchValue: '',
+                allTreeItems: [],
+                filteredTreeItems: []
+            }
+        },
+        computed: {
+            treeItems() {
+                if (this.searchValue === '') {
+                    return this.allTreeItems;
+                } else {
+                    return this.filteredTreeItems;
+                }
+            }
+        },
+        methods: {
+            getAllChildren() {
+                this.openmct.objects.get('ROOT')
+                    .then(root => this.openmct.composition.get(root).load())
+                    .then(children => {
+                        this.allTreeItems = children.map(c => {
+                                return {
+                                    id: this.openmct.objects.makeKeyString(c.identifier),
+                                    object: c,
+                                    objectPath: [c]
+                            };
+                        });
+                    });
+            },
+            getFilteredChildren() {
+                this.searchService.query(this.searchValue).then(children => {
+                    this.filteredTreeItems = children.hits.map(child => {
+                        let objectPath = child.object.getCapability('context')
```

Comment:
I just found an issue in testing. The context capability returned the 'ROOT' object, which we don't need for contextualization, and in fact it results in broken navigation. The solution is simple though, just `slice` it from the return context path, eg.
```javascript
let objectPath = child.object.getCapability('context').getPath()
  .slice(1)
  .map(oldObject => oldObject.useCapability('adapter'))
  .reverse();
//...
```